### PR TITLE
Fix mutable defaults

### LINKED ISSUE #355: Several functions have mutable default arguments
I came across a function in pygsti version 0.9.11.2 that has a mutable default argument. This can lead to nasty bugs, as described [here](https://docs.python-guide.org/writing/gotchas/). Once I found the first bug of this kind, I ran some searches and found several more. See below for a list.

#### Search results

I ran a regex search for strings of the form ``=[]<characters here>):`` [1] and found the following hits:
image

If I search for strings of the form ``={}<characters here>):`` [2], I get 
image

#### Footnotes

[Footnote 1] ``<insert characters here>`` is a placeholder for any character sequence that excludes new lines. The actual regex I used was ``=\[\].+\):``. Note that this doesn't capture all possible cases where we have mutable default arguments, since it will miss cases of multi-line function definitions where the mutable argument isn't on the last line.

[Footnote 2] The actual regex I used was ``=\{\}.+\):``.

You can search for both patterns I checked with ``=[\{\[][\}\]].+\):``.

### PR DESCRIPTION
**Pull request by rileyjmurray**
This PR should resolve #355.

When I tested these changes, I found that some code still called ``pygsti.obj``, even though this subpackage was renamed to ``pygsti.baseobjs`` some time ago. This PR also replaces all instances of ``pygsti.obj.`` with ``pygsti.baseobjs.`` to fix that.

### CODE REVIEW

File: pygsti/extras/drift/driftreport.py [hunk 1]
```diff
@@ -99,13 +99,15 @@ class PowerSpectraPlot(_ws.WorkspacePlot):
...
-    def __init__(self, ws, results, spectrumlabel={}, detectorkey=None,
+    def __init__(self, ws, results, spectrumlabel=None, detectorkey=None,
                  showlegend=False, scale=1.0):
         """
         todo
         """
         super(PowerSpectraPlot, self).__init__(ws, self._create, results,
                                                spectrumlabel, detectorkey, showlegend, scale)
+        if spectrumlabel is None:
+            spectrumlabel = {}
```
Comment (rileyjmurray) [inline pygsti/extras/drift/driftreport.py:110]: This needs to be before the call to ``super``.

**Comment by rileyjmurray** [comment]
IMO: ready to merge, once https://github.com/pyGSTio/pyGSTi/pull/357#discussion_r1364078892 is addressed.

File: pygsti/extras/drift/core.py
[the remark this answers is not in the document]
Comment (sserita) [reply pygsti/extras/drift/core.py:1]: Yes we can delete this. This is actually indicative of a larger pattern we will see during spring cleaning, where things are defined in extras and slowly migrated into PyGSTi over time. In this case, basically only the StabilityAnalyzer and driftreport.py are used directly - we should probably shift these into pygsti.protocol and pygsti.report respectively and eliminate pygsti.extras.drift entirely. But I'll leave that for a future issue/PR.

**Comment by rileyjmurray** [comment]
@sserita: ready to merge as far as I'm concerned.

**Comment by sserita** [comment]
I agree that this is ready. I'll wait until the tests on beta pass from the other big PR before merging this in though (even after these are passing).

**Comment by rileyjmurray** [comment]
@sserita, did the tests on beta pass for the other PR you mentioned? If so, can you merge this PR?

**Pull request merged**